### PR TITLE
Added DETR ResNet50 model

### PR DESCRIPTION
**Pull request by Anna-Mironova**

### CODE REVIEW

File: demos/common/python/models/detr.py [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+        self.n, self.c, self.h, self.w = self.net.input_info[self.image_blob_name].input_data.shape
+        assert self.c == 3, "Expected 3-channel input"
+
+    def _parse_outputs(self):
+        bboxes_blob_name = None
+        scores_blob_name = None
+        for name, layer in self.net.outputs.items():
+            if layer.shape[2] == 4:
+                bboxes_blob_name = name
+            elif layer.shape[2] == 92:
```
Comment (eaidova) [inline demos/common/python/models/detr.py:46]: this restriction does not allow you to run model versions trained with another number of classes

File: models/public/detr-resnet50/README.md [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+    - `B` - batch size
+    - `A` - number of detected anchors
+
+    For each detection, the description has the format: [`x`, `y`, `w`, `h`], where:
+
+    - (`x`, `y`) - coordinates of the bounding box center(coordinates are in normalized format, in range [0, 1])
+    - `w` - width of bounding box(values are in normalized format, in range [0, 1])
+    - `h` - height of bounding box(values are in normalized format, in range [0, 1])
+
+2. Scores, name: `scores`, shape - `1, 100, 92`. Contains scores for background and 91 object classes which are listed in `<omz_dir>/data/dataset_classes/coco_91cl_bkgr.txt` file.
```
Comment (eaidova) [inline models/public/detr-resnet50/README.md:69]: but in demo code you disable the last class, does it means that background (or no_detection in DETR terminology is the last class) and if you run demo with provided class names, names will be incorrect?

File: demos/tests/cases.py [hunk 3]
```diff
@@ -841,6 +841,16 @@ def single_option_cases(key, *args):
             ),
             TestCase(options={'-at': 'yolov4', '-m': ModelArg('yolo-v4-tf')}),
             TestCase(options={'-at': 'yolov4', '-m': ModelArg('yolo-v4-tiny-tf')}),
+            *combine_cases(
+                TestCase(options={'--architecture_type': 'detr'}),
+                [
+                    TestCase(options={'-m': ModelArg('detr-resnet50')}),
```
Comment (eaidova) [inline demos/tests/cases.py:847]: please add test for IR too

File: models/public/detr-resnet50/accuracy-check.yml [hunk 4]
```diff
@@ -0,0 +1,22 @@
+models:
+  - name: detr-resnet50
```
Comment (eaidova) [inline models/public/detr-resnet50/accuracy-check.yml:2]: please provide section for onnx runtime

File: tools/accuracy_checker/accuracy_checker/adapters/detection.py [hunk 5]
```diff
@@ -897,9 +897,9 @@ def box_cxcywh_to_xyxy(x):
                  (x_c + 0.5 * w), (y_c + 0.5 * h)]
             return b
 
-        def softmax(x):
-            exp_x = np.exp(x)
-            return exp_x / np.sum(exp_x)
+        def softmax(logits):
```
Comment (eaidova) [inline tools/accuracy_checker/accuracy_checker/adapters/detection.py:900]: why do you need this?

File: models/public/detr-resnet50/README.md [hunk 6]
```diff
@@ -0,0 +1,104 @@
+# detr-resnet50
+
+## Use Case and High-Level Description
+
+The `detr-resnet50` model is a model of the new framework, called DEtection TRansformer or DETR with ResNet50 backbone for object detection task. DETR predicts all objects at once, and is trained end-to-end with a set loss function which performs bipartite matching between predicted and ground-truth objects. DETR simplifies the detection pipeline by dropping multiple hand-designed components that encode prior knowledge, like spatial anchors or non-maximal suppression.
```
Comment (eaidova) [inline models/public/detr-resnet50/README.md:5]:
```suggestion
The `detr-resnet50` model is one from DEtection TRansformer (DETR) models family, which consider object detection as a direct set prediction problem. The model has ResNet50 backbone and pretrained on COCO dataset for solving object detection task. DETR predicts all objects at once, and is trained end-to-end with a set loss function which performs bipartite matching between predicted and ground-truth objects. DETR simplifies the detection pipeline by dropping multiple hand-designed components that encode prior knowledge, like spatial anchors or non-maximal suppression.
```

File: models/public/detr-resnet50/accuracy-check.yml (hunk 4 above)
Comment (Anna-Mironova) [reply models/public/detr-resnet50/accuracy-check.yml:27]: Done

File: models/public/detr-resnet50/README.md (hunk 6 above)
Comment (Anna-Mironova) [reply models/public/detr-resnet50/README.md]: Done

File: tools/accuracy_checker/accuracy_checker/adapters/detection.py (hunk 5 above)
Comment (Anna-Mironova) [reply tools/accuracy_checker/accuracy_checker/adapters/detection.py:900]: Softmax is applied along entire scores vector in the previous implementation, but in the original version softmax is applied to values along last dim https://colab.research.google.com/github/facebookresearch/detr/blob/colab/notebooks/detr_demo.ipynb#scrollTo=BiwSmd2i-Wkf

**Comment by Anna-Mironova** [comment]
Jenkins please retry a build

File: demos/common/python/models/detr.py [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+        x_mins = x_mins * meta['original_shape'][1]
+        y_mins = y_mins * meta['original_shape'][0]
+        x_maxs = x_maxs * meta['original_shape'][1]
+        y_maxs = y_maxs * meta['original_shape'][0]
+
+        scores = self.softmax(scores)
+        labels = np.argmax(scores[:, :-1], axis=-1)
+        det_scores = np.max(scores[:, :-1], axis=-1)
+
+        return [Detection(*det) for det in zip(x_mins, y_mins, x_maxs, y_maxs, det_scores, labels)]
```
Comment (Wovchena) [inline demos/common/python/models/detr.py:81]: scores are no longer filtered by `draw_detections()` and `print_raw_results()`. You should do it here. See https://github.com/openvinotoolkit/open_model_zoo/pull/2551

Comment (Anna-Mironova) [reply demos/common/python/models/detr.py]: Done, thanks

**Review by eaidova — APPROVED** [review]
